smack-extensions/src/main/java/org/jivesoftware/smackx/pubsub/Subscription.java

```diff
@@ -56,6 +56,17 @@ public Subscription(String subscriptionJid, String nodeId)
         this(subscriptionJid, nodeId, null, null);
     }
 
+    /**
+     * Construct a subscription change request to the specified state.
+     *
+     * @param subscriptionJid The subscriber JID
+     * @param state The requested new state
+     */
+    public Subscription(String subscriptionJid, State state)
```

Comment:
I'd suggest using `BareJid` instead of `String` as the subsriptionJid type.

Reply:
Understood. The main issue is that the JID is held as a String internally, and all the other ctors use String.
I could take a BareJid parameter and convert it to a String... One tiny step in the right direction !
Converting the class to use a Jid consistently would be a breaking change. I could keep the String parameters, and throw an exception if they fail to convert to the correct type of Jid for the given circumstance (assuming that a Subscription doesn't always require a BareJid - I would need to look that up :-)
What do you think ?

Reply:
Honestly I'm not sure whats the right way to go. Ping @Flowdalic.

Reply:
> Converting the class to use a Jid consistently would be a breaking change.

I just pushed this change to master, which means your PR needs to adopt that.